internal/controller/storageclient_controller.go

```diff
@@ -156,8 +210,77 @@ func (r *storageClientReconcile) reconcile() error {
 	return nil
 }
 
+func (r *storageClientReconcile) isCrdBeingWatched(crdName string) bool {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return r.crdsBeingWatched[crdName]
+}
+
+func (r *storageClientReconcile) markCrdBeingWatched(crdName string) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.crdsBeingWatched[crdName] = true
+}
+
+func (r *storageClientReconcile) reconcileDynamicWatches() error {
+	for idx := range crdsToWatch {
```

Comment:
There is no point in running in a loop if each iteration need specific logic based on the name.
Just have normal non iterative code to set up the watch. Similar to what we have in SetupWithManager

Reply:
done.